src/libs/ajax/file-browser-providers/GCSFileBrowserProvider.ts

```diff
@@ -0,0 +1,101 @@
+import { Ajax } from 'src/libs/ajax'
+import FileBrowserProvider from 'src/libs/ajax/file-browser-providers/FileBrowserProvider'
+import { GCSItem } from 'src/libs/ajax/GoogleStorage'
+import IncrementalResponse from 'src/libs/ajax/IncrementalResponse'
+
+
+export interface GCSFileBrowserProviderParams {
+  bucket: string
+  project: string
+  pageSize?: number
+}
+
+type GCSFileBrowserProviderGetPageParams<T> = {
+  isFirstPage: boolean
+  pageToken?: string
+  pendingItems?: T[]
+  prefix: string
+  previousItems?: T[]
+  signal: any
+} & ({
+  itemsOrPrefixes: 'items'
+  mapItemOrPrefix: (item: GCSItem) => T
+} | {
+  itemsOrPrefixes: 'prefixes'
+  mapItemOrPrefix: (prefix: string) => T
+})
+
+const GCSFileBrowserProvider = ({ bucket, project, pageSize = 1000 }: GCSFileBrowserProviderParams): FileBrowserProvider => {
```

Comment:
lets add a jsdoc comment explaining the general use of this provider, and the fact that it incrementally grows the result set with each call to getNextPage

Reply:
Added a comment to IncrementalResponse in https://github.com/DataBiosphere/terra-ui/pull/3468/commits/756c8fe00e6bd4b7d6091b2edebb9169881b44ac.